extension-manifest-v3/src/background/custom-filters.js

```diff
@@ -8,10 +8,102 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0
  */
-import { parseFilters } from '@cliqz/adblocker';
+import { store } from 'hybrids';
+import {
+  parseFilters,
+  detectFilterType,
+  FilterType,
+  CosmeticFilter,
+} from '@cliqz/adblocker';
 
+import {
+  createDocumentConverter,
+  createOffscreenConverter,
+} from '/utils/dnr-converter.js';
 import * as engines from '/utils/engines.js';
 
+import Options, { observe } from '/store/options.js';
+import CustomFilters from '/store/custom-filters.js';
+
+const convert =
+  __PLATFORM__ !== 'safari' && __PLATFORM__ !== 'firefox'
+    ? createOffscreenConverter()
+    : createDocumentConverter();
+
+class TrustedScriptletError extends Error {}
+
+// returns a scriptlet with encoded arguments
+// returns undefined if not a scriptlet
+// throws if scriptlet cannot be trusted
+function fixScriptlet(filter, trustedScriptlets) {
+  const cosmeticFilter = CosmeticFilter.parse(filter);
+
+  if (
+    !cosmeticFilter ||
+    !cosmeticFilter.isScriptInject() ||
+    !cosmeticFilter.selector
+  ) {
+    return null;
+  }
+
+  const parsedScript = cosmeticFilter.parseScript();
+
+  if (!parsedScript || !parsedScript.name) {
+    return null;
+  }
+
+  if (
+    !trustedScriptlets &&
+    (parsedScript.name === 'rpnt' ||
+      parsedScript.name === 'replace-node-text' ||
+      parsedScript.name.startsWith('trusted-'))
+  ) {
+    throw new TrustedScriptletError();
+  }
+
+  const [front] = filter.split(`#+js(${parsedScript.name}`);
+  const args = parsedScript.args.map((arg) => encodeURIComponent(arg));
+  return `${front}#+js(${[parsedScript.name, ...args].join(', ')})`;
+}
+
+function parseInput(text = '', { trustedScriptlets }) {
```

Comment:
IMHO, the old name "parseFilters" was more descriptive than "parseInput".

Reply:
The function is not required anymore - the only part with fixing scriptlets will be moved, after discussion with @chrmod